ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,564 @@
+#include <tensor_common.h>
+
+#include <core/session/onnxruntime_cxx_api.h>
+
+
+/**
+ * @brief Macro for debug mode.
+ */
+#ifndef DBG
+#define DBG FALSE
+#endif
+
+
+namespace nnstreamer
+{
+namespace tensor_filter_onnxruntime
+{
+extern "C" {
+void init_filter_onnxruntime (void) __attribute__ ((constructor));
+void fini_filter_onnxruntime (void) __attribute__ ((destructor));
+}
+
+/** @brief tensor-filter-subplugin concrete class for onnxruntime */
+class onnxruntime_subplugin final : public tensor_filter_subplugin
+{
+  private:
+  bool empty_model;
+  char *model_path; /**< The model *.onnx file */
+
+  GstTensorsInfo inputInfo; /**< Input tensors metadata */
+  GstTensorsInfo outputInfo; /**< Output tensors metadata */
+
+  Ort::Session session;
+  Ort::SessionOptions sessionOptions;
+  Ort::Env env;
+  Ort::MemoryInfo memInfo;
+
+  std::size_t input_num_tensors;
+  std::vector<const char *> input_names;
+  std::vector<std::vector<int64_t>> input_shapes;
+  std::vector<ONNXTensorElementDataType> input_types;
+  std::vector<Ort::AllocatedStringPtr> input_names_allocated_strings;
+  std::vector<Ort::Value> input_tensors; /**< Input tensor from model */
+
+  std::size_t output_num_tensors;
+  std::vector<const char *> output_names;
+  std::vector<std::vector<int64_t>> output_shapes;
+  std::vector<ONNXTensorElementDataType> output_types;
+  std::vector<Ort::AllocatedStringPtr> output_names_allocated_strings;
+  std::vector<Ort::Value> output_tensors; /**< Output tensor from model */
+
+  static const char *name;
+  static onnxruntime_subplugin *registeredRepresentation;
+
+  void cleanup ();
+  int getTensorDim (tensor_dim &dim, std::vector<int64_t> shapes, size_t numDims);
+  int getTensorType (ONNXTensorElementDataType _type, tensor_type *type);
+
+  public:
+  static void init_filter_onnxruntime ();
+  static void fini_filter_onnxruntime ();
+
+  onnxruntime_subplugin ();
+  ~onnxruntime_subplugin ();
+
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+};
+
+const char *onnxruntime_subplugin::name = "onnxruntime";
+
+/**
+ * @brief Constructor for onnxruntime_subplugin.
+ */
+onnxruntime_subplugin::onnxruntime_subplugin ()
+    : tensor_filter_subplugin (), empty_model (true), model_path (nullptr),
+      session (nullptr), sessionOptions (nullptr), env (nullptr),
+      memInfo (nullptr), input_num_tensors (0U), input_names (), input_shapes (),
+      input_types (), input_names_allocated_strings (), input_tensors (),
+      output_num_tensors (0U), output_names (), output_shapes (),
+      output_types (), output_names_allocated_strings (), output_tensors ()
```

Comment:
```suggestion
    : empty_model {true}, model_path {nullptr},
      session {nullptr}, sessionOptions {nullptr}, env {nullptr},
      memInfo {nullptr}, input_num_tensors {0U}, input_names {}, input_shapes {},
      input_types {}, input_names_allocated_strings {}, input_tensors {},
      output_num_tensors {0U}, output_names {}, output_shapes {},
      output_types {}, output_names_allocated_strings {}, output_tensors {}
```

Q1: Is ```tensor_filter_subplugin ()``` necessary?
Q2: Why are those empty initializers required? I think they will be already initialized by default.

Reply:
I'm not sure these initializers are necessary. But, i thinks, initializing values with defaults avoid potential issues related to uninitialized variables.

Reply:
Actually, that is unnecessary:
```C++
$ g++ cppclass.cpp 
$ ./a.out 
base();
derived();
~derived();
~base();
$ cat cppclass.cpp 
#include <stdio.h>

class base {
public:
	base() { printf("base();\n"); }
	~base() { printf("~base();\n"); }
};
class derived : public base {
public:
	derived() { printf("derived();\n"); }
	~derived() { printf("~derived();\n"); }
};
int main() {
	derived d;
	return 0;
}
$ 
```

Reply:
Thanks. updated it.